fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -261,6 +274,50 @@ private HTTPReturnPreference computeReturnPref(ServletRequest request, HTTPHandl
         return returnPref;
     }
 
+    /**
+     * Check that FHIR version parameters in the HTTP headers are valid.
+     * @param requestHeaders the headers
+     * @throws FHIRRestServletRequestException
+     */
+    void checkFhirVersionParameter(Map<String, List<String>> requestHeaders) throws FHIRRestServletRequestException {
+        Map<String, Integer> headerStatusMap = new LinkedHashMap<>();
+        // 415 Unsupported Media Type is the appropriate response when the client posts a format that is not supported to the server.
+        headerStatusMap.put(HttpHeaders.CONTENT_TYPE, HttpServletResponse.SC_UNSUPPORTED_MEDIA_TYPE);
+        // 406 Not Acceptable is the appropriate response when the Accept header requests a format that the server does not support.
+        headerStatusMap.put(HttpHeaders.ACCEPT, HttpServletResponse.SC_NOT_ACCEPTABLE);
+
+        for (String headerName : headerStatusMap.keySet()) {
+            String fhirVersion = null;
+            for (String headerValue : requestHeaders.getOrDefault(headerName, Collections.emptyList())) {
+                if (headerValue != null) {
+                    for (String headerValueElement : headerValue.split(",")) {
+                        Map<String, String> parameters = MediaType.valueOf(headerValueElement).getParameters();
+                        if (parameters != null) {
+                            for (Map.Entry<String, String> parameter : parameters.entrySet()) {
+                                if (FHIRMediaType.FHIR_VERSION_PARAMETER.equalsIgnoreCase(parameter.getKey())) {
+                                    String curFhirVersion = parameter.getValue();
+                                    if (curFhirVersion != null && !FHIRMediaType.SUPPORTED_FHIR_VERSIONS.contains(curFhirVersion)) {
+                                        throw new FHIRRestServletRequestException("Invalid '" + FHIRMediaType.FHIR_VERSION_PARAMETER
+                                            + "' parameter value in '" + headerName + "' header; the following FHIR versions are supported: "
+                                                + FHIRMediaType.SUPPORTED_FHIR_VERSIONS, headerStatusMap.get(headerName));
```

Comment:
thats a lot of nesting!

Reply:
do we need the null checks on every level?  if yes, then maybe there's not much we can do about it.

Reply:
I was able to remove a couple levels of nesting.